Ingester panic while replaying WAL and cutting block

**Describe the bug**
Ingesters panic while replaying the WAL and cutting new blocks during start-up.

```
level=info ts=2022-01-10T19:01:35.74658104Z caller=app.go:270 msg="Tempo started"                                                                                                                                                                                            
level=info ts=2022-01-10T19:01:35.747246377Z caller=lifecycler.go:601 msg="existing entry found in ring" state=ACTIVE tokens=128 ring=ingester                                                                                                                               
level=info ts=2022-01-10T19:01:39.105440637Z caller=flush.go:243 msg="completing block" userid=<user-id> blockID=<block-id>                                                                                                                                   
level=info ts=2022-01-10T19:01:44.986318804Z caller=flush.go:243 msg="completing block" userid=<user-id> blockID=<block-id>                                                                                                                                   
level=info ts=2022-01-10T19:01:51.293651144Z caller=flush.go:250 msg="block completed" userid=<user-id> blockID=<block-id>    duration=6.307331388s                                                                                                           
level=info ts=2022-01-10T19:01:51.323608067Z caller=flush.go:299 msg="flushing block" userid=<user-id> block=<block-id>                                                                                                                                    
panic: runtime error: index out of range [25314] with length 25314                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                             
goroutine 453 [running]:                                                                                                                                                                                                                                                     
github.com/pierrec/lz4/v4/internal/lz4block.(*Compressor).CompressBlock(0xc0411b4000, {0xc0431a4000, 0x62e2, 0x100000}, {0xc0ce880000, 0x62e2, 0x100000})                                                                                                                    
    /drone/src/vendor/github.com/pierrec/lz4/v4/internal/lz4block/block.go:193 +0xb9e                                                                                                                                                                                        
github.com/pierrec/lz4/v4/internal/lz4block.CompressBlock({0xc0431a4000, 0x62e2, 0x100000}, {0xc0ce880000, 0x62e2, 0x100000})                                                                                                                                                
    /drone/src/vendor/github.com/pierrec/lz4/v4/internal/lz4block/block.go:91 +0x8b                                                                                                                                                                                          
github.com/pierrec/lz4/v4/internal/lz4stream.(*FrameDataBlock).Compress(0xc06ca86620, 0xc060352090, {0xc0431a4000, 0xc056f9f9c0, 0x1197f3c}, 0x603521b0)                                                                                                                     
    /drone/src/vendor/github.com/pierrec/lz4/v4/internal/lz4stream/block.go:221 +0x85                                                                                                                                                                                        
github.com/pierrec/lz4/v4.(*Writer).write(0xc0603521b0, {0xc0431a4000, 0x0, 0x0}, 0x0)                                                                                                                                                                                       
    /drone/src/vendor/github.com/pierrec/lz4/v4/writer.go:128 +0x1af                                                                                                                                                                                                         
github.com/pierrec/lz4/v4.(*Writer).Close(0xc0603521b0)                                                                                                                                                                                                                      
    /drone/src/vendor/github.com/pierrec/lz4/v4/writer.go:162 +0xb3                                                                                                                                                                                                          
github.com/grafana/tempo/tempodb/encoding/v2.(*dataWriter).CutPage(0xc0581f5680)                                                                                                                                                                                             
    /drone/src/tempodb/encoding/v2/data_writer.go:60 +0x82                                                                                                                                                                                                                   
github.com/grafana/tempo/tempodb/encoding.(*bufferedAppender).flush(0xc0581f56d0)                                                                                                                                                                                            
    /drone/src/tempodb/encoding/appender_buffered.go:113 +0x3b                                                                                                                                                                                                               
github.com/grafana/tempo/tempodb/encoding.(*bufferedAppender).Append(0xc0581f56d0, {0xc097cf6000, 0xc06bcd5f50, 0x284}, {0xc09b500000, 0x5b4f631e2940be2f, 0x2177cec54c37dc8d})                                                                                              
    /drone/src/tempodb/encoding/appender_buffered.go:55 +0xe5                                                                                                                                                                                                                
github.com/grafana/tempo/tempodb/encoding.(*StreamingBlock).AddObject(0xc06031f5c0, {0xc097cf6000, 0x10, 0x112284}, {0xc09b500000, 0x27f3cf8, 0xc0003942f0})                                                                                                                 
    /drone/src/tempodb/encoding/streaming_block.go:64 +0x42                                                                                                                                                                                                                  
github.com/grafana/tempo/tempodb.(*readerWriter).CompleteBlockWithBackend(0xc040f980c0, {0x27e33d0, 0xc00005c030}, 0x0, {0x27a0be0, 0x3b70498}, {0x280d788, 0xc0003942e0}, {0x27f3cf8, 0xc0003942f0})                                                                        
    /drone/src/tempodb/tempodb.go:244 +0x3b4                                                                                                                                                                                                                                 
github.com/grafana/tempo/modules/ingester.(*instance).CompleteBlock(0xc000632ea0, {0x2f, 0xbe, 0x40, 0x29, 0x1e, 0x63, 0x4f, 0x5b, 0x8d, ...})                                                                                                                               
    /drone/src/modules/ingester/instance.go:304 +0x163                                                                                                                                                                                                                       
github.com/grafana/tempo/modules/ingester.(*Ingester).handleComplete(0xc000250900, 0xc04b102320)                                                                                                                                                                             
    /drone/src/modules/ingester/flush.go:249 +0x25c                                                                                                                                                                                                                          
github.com/grafana/tempo/modules/ingester.(*Ingester).flushLoop(0xc000250900, 0x1)                                                                                                                                                                                           
    /drone/src/modules/ingester/flush.go:202 +0xc7                                                                                                                                                                                                                           
created by github.com/grafana/tempo/modules/ingester.New                                                                                                                                                                                                                     
    /drone/src/modules/ingester/ingester.go:81 +0x1e9
```

**To Reproduce**
Running image [grafana/tempo:main-ef4213c](https://hub.docker.com/layers/grafana/tempo/main-ef4213c/images/sha256-2ca424393d617f07b44c52980ae1e469db583ee502e2944746deea46289a66c6?context=explore)

**Expected behavior**
Ingesters does not panic

**Environment:**
 - Infrastructure: Kubernetes
 - Deployment tool: jsonnet

--- a/go.mod
+++ b/go.mod
@@ -44,7 +44,7 @@ require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/zipkinreceiver v0.40.0
 	github.com/opentracing-contrib/go-grpc v0.0.0-20210225150812-73cb765af46e
 	github.com/opentracing/opentracing-go v1.2.0
-	github.com/pierrec/lz4/v4 v4.1.3
+	github.com/pierrec/lz4/v4 v4.1.12
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.11.0
 	github.com/prometheus/client_model v0.2.0

